File: java/problems/p0009_palindrome_number/Solution.java
package problems.p0009_palindrome_number;

class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        String x_str = String.valueOf(x);

        for (int i = 0; i < x_str.length() / 2; i++) {
            if (x_str.charAt(i) != x_str.charAt(x_str.length() - 1 - i)) {
                return false;
            }
        }

        return true;
    }
}

File: typescript/problems/p0009_palindrome_number/solution.ts
function isPalindrome(x: number): boolean {
  if (x < 0) return false;

  const x_str = x.toString();

  for (let i = 0; i < x_str.length / 2; i++) {
    if (x_str[i] !== x_str[x_str.length - 1 - i]) {
      return false;
    }
  }

  return true;
}

export default isPalindrome;
